src/Plugin/Plugins/TempHumidPlugin.cpp

```diff
@@ -0,0 +1,259 @@
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+/* Initialize image path for temperature (scale) icon. */
+const char* TempHumidPlugin::IMAGE_PATH_TEMP_ICON      = "/images/temp.bmp";
+
+/* Initialize image path for humidity (drop) icon. */
+const char* TempHumidPlugin::IMAGE_PATH_HUMID_ICON     = "/images/drop.bmp";
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+void TempHumidPlugin::setSlot(const ISlotPlugin* slotInterf)
+{
+    m_slotInterf = slotInterf;
+    return;
+}
+
+void TempHumidPlugin::active(IGfx& gfx)
+{
+    lock();
+
+    // set time to show page - either 10s or slot_time / 4
+    // read here because otherwise we do not get config changes in slot_time
+    if (nullptr == m_slotInterf) {
+        m_pageTime = 10000U;
+    } else {
+        m_pageTime = m_slotInterf->getDuration() / 4;
+    }
+
+    gfx.fillScreen(ColorDef::BLACK);
+
+    if (nullptr == m_iconCanvas)
+    {
+        m_iconCanvas = new Canvas(ICON_WIDTH, ICON_HEIGHT, 0, 0);
+
+        if (nullptr != m_iconCanvas)
+        {
+            (void)m_iconCanvas->addWidget(m_bitmapWidget);
+
+            /* Load icon from filesystem. */
+            (void)m_bitmapWidget.load(FILESYSTEM, IMAGE_PATH_TEMP_ICON);
+
+            m_iconCanvas->update(gfx);
+        }
+    }
+    else
+    {
+        m_iconCanvas->update(gfx);
+    }
+
+    if (nullptr == m_textCanvas)
+    {
+        m_textCanvas = new Canvas(gfx.getWidth() - ICON_WIDTH, gfx.getHeight(), ICON_WIDTH, 0);
+
+        if (nullptr != m_textCanvas)
+        {
+            (void)m_textCanvas->addWidget(m_textWidget);
+
+            m_textCanvas->update(gfx);
+        }
+    }
+    else
+    {
+        m_textCanvas->update(gfx);
+    }
+
+    unlock();
+
+    return;
+}
+
+void TempHumidPlugin::update(IGfx& gfx)
+{
+    bool            showPage    = false;
+    char            tmp[6]      = "";     // holds the display value
+
+    if (false == m_timer.isTimerRunning())
+    {
+        m_timer.start(m_pageTime);
+        showPage = true;
+    }
+    else if (true == m_timer.isTimeout())
+    {
+        ++m_page;
+        m_page %= 2U;
```

Comment:
2U is magic here ... see in the switch/case below too.
Proposal:
Use a enumeration like
```
enum Page
{
    PAGE_0 = 0,
    PAGE_1,
    PAGE_MAX
}
```

PAGE_MAX gives you the nunber of pages, which are available and the others you can use in the switch/case.

Reply:
Good idea, implemented as proposed.